service-workers/service-worker/ServiceWorkerGlobalScope/extendable-message-event.https.html

```diff
@@ -5,222 +5,221 @@
-          return wait_for_state(t, registration.installing, 'activated');
-        })
-      .then(function() {
-          // A controlled frame is necessary for keeping a waiting worker.
-          return with_iframe(scope);
-        })
-      .then(function(frame) {
-          add_completion_callback(function() { frame.remove(); });
-          return navigator.serviceWorker.register(script2, {scope: scope});
-        })
-      .then(function(r) {
-          return wait_for_state(t, r.installing, 'installed');
-        })
-      .then(function() {
-          var results = [];
-          var saw_message = new Promise(function(resolve) {
-              navigator.serviceWorker.onmessage = function(event) {
-                switch (event.data.type) {
-                  case 'record':
-                    results.push(event.data.results);
-                    break;
-                  case 'finish':
-                    resolve(results);
-                    break;
-                }
-              };
-            });
-          registration.active.postMessage({type: 'start'});
-          return saw_message;
-        })
-      .then(function(results) {
-          assert_equals(results.length, 2);
-
-          var expected_ping = {
-              constructor: { name: 'ExtendableMessageEvent' },
-              origin: location.origin,
-              lastEventId: '',
-              source: {
-                  constructor: { name: 'ServiceWorker' },
-                  scriptURL: normalizeURL(script1),
-                  state: 'activated'
-              },
-              ports: []
-          };
-          assert_equals(results[0].pingOrPong, 'ping');
-          ExtendableMessageEventUtils.assert_equals(
-              results[0].event, expected_ping
-          );
-
-          var expected_pong = {
-              constructor: { name: 'ExtendableMessageEvent' },
-              origin: location.origin,
-              lastEventId: '',
-              source: {
-                  constructor: { name: 'ServiceWorker' },
-                  scriptURL: normalizeURL(script2),
-                  state: 'installed'
-              },
-              ports: []
-          };
-          assert_equals(results[1].pingOrPong, 'pong');
-          ExtendableMessageEventUtils.assert_equals(
-              results[1].event, expected_pong
-          );
-        });
-  }, 'Post extendable messages among service workers');
+promise_test(async t => {
+  const SCRIPT = 'resources/extendable-message-event-worker.js';
+  const SCOPE = 'resources/scope/extendable-message-event-from-toplevel';
+  const registration = await service_worker_unregister_and_register(
+    t,
+    SCRIPT,
+    SCOPE
+  );
+
+  t.add_cleanup(async() => {
+    if (registration) await registration.unregister();
+  });
+
+  await wait_for_state(t, registration.installing, 'activated');
+  const saw_message = new Promise(function(resolve) {
+    navigator.serviceWorker.onmessage = function(event) { 
```

Comment:
The linter seems to complain here:
ERROR:lint:service-workers/service-worker/ServiceWorkerGlobalScope/extendable-message-event.https.html:23: Whitespace at EOL (TRAILING WHITESPACE)

https://taskcluster-artifacts.net/Ze_zL2WuQz-rAD9J9qPNvQ/0/public/logs/live_backing.log

Reply:
Many thanks for you point out the mistake, correct it in latest commit.